docs/src/main/mdoc/consumers.md

```diff
@@ -0,0 +1,314 @@
+---
+id: consumers
+title: Consumers
+---
+
+Consumers support subscribing to topics, record streaming and deserialization, as well as miscellaneous utility functionality, such as seeking to offsets, or checking what the end offsets are for a topic. Consumers make use of the [Java Kafka consumer][java-kafka-consumer], which becomes especially important for [settings](#settings). For consumer implementation details, refer to the [technical details](technical-details.md) section.
+
+The following imports are assumed throughout this page.
+
+```scala mdoc:silent
+import cats.effect._
+import cats.implicits._
+import fs2.kafka._
+import scala.concurrent.duration._
+```
+
+## Deserializers
+
+[`Deserializer`][deserializer] describes functional composable deserializers for record keys and values. We generally require two deserializers: one for the record key and one for the record value. Deserializers are provided implicitly for many standard library types, including:
+
+- `Array[Byte]`, `Double`, `Float`, `Int`, `Long`, `Short`, `String`, and `UUID`.
+
+There are also deserializers for types which carry special meaning:
+
+- `Option[A]` to deserialize occurrances of `null` as `None`, and
+
+- `Unit` to ignore the serialized bytes and always use `()`.
+
+For more involved types, we need to resort to custom deserializers.
+
+### Custom Deserializers
+
+`Deserializer[F[_], A]` describes a function `Array[Byte] => F[A]`, while also having access to the topic name and record [`Headers`][headers]. There are many [functions][deserializer$] available for creating custom deserializers, with the most basic one being `instance`, which simply creates a deserializer from a provided function.
+
+```scala mdoc:silent
+Deserializer.instance {
+  (topic, headers, bytes) =>
+    IO.pure(bytes.dropWhile(_ == 0))
+}
+```
+
+If the deserializer only needs access to the bytes, like in the case above, use `lift`.
+
+```scala mdoc:silent
+Deserializer.lift(bytes => IO.pure(bytes.dropWhile(_ == 0)))
+```
+
+To support different deserializers for different topics, use `topic`.
```

Comment:
Maybe be more explicit that we are pattern matching on topic names? 
Something like:
"To support different deserializers for different topics, use `topic`, which will allow to pattern match on the topic names."

Reply:
I went with:
> To support different deserializers for different topics, use `topic` to pattern match on the topic name.